docs/HTTPSCertificates.md

```diff
@@ -0,0 +1,24 @@
+# Managing nginx HTTPS Certificates
+
+nginx needs to provide secure HTTP communication between clients and servers. This means that the server needs to be able to verify itself to the client so that the client is confident it's sending messages to the correct place, and that the client can communicate in a secure fashion with this server. nginx achieves this through HTTPS (HTTP over SSL). This involves the server sending a signed certifcate (or public key) to the client, who is able to verify the signature. The client may also request the certificate authority (CA) certificate signing request (CSR) which was used to sign the certificate. Once the client has verified the signed certificate, it can use it to securely send a means of encryption for subsequent messages to the server. The server can read this using the private key, and the server and client can now exchange information securely. There are three options for generating the secure certificate.
+
+## Self-signed certificate
+
+This requires the client accepting un-trusted certificates, and is the default option for the clearwater-nginx package. It involves the server creating its own CSR, which it uses to sign its certificate. Currently the post-install script for clearwater-nginx handles this using the openssl library.
+
+    openssl req -nodes -sha256 -newkey rsa:2048 -keyout nginx.key -out nginx.csr -config nginx_openssl_config
+    openssl x509 -sha256 -req -in nginx.csr -signkey nginx.key -out nginx.crt
+
+The first command creates a private key and a CSR from a very basic default set of config. The second command creates a signed certificate using the private key and the CSR. Since this is all done at start of life, clients won't be able to verify the certificate, and so will have to accept un-trusted certificates, which is not secure.
```

Comment:
Yes, I know, I'll make that clear if you like.

It depends what they're using nginx for, that's why I wrote the memento article. Obviously I could say call them nginx.* and put them on your server at /etc/nginx/ssl or wherever it is but that felt kind of random.